src/app/identity.service.ts

```diff
@@ -97,14 +98,36 @@ export class IdentityService {
     });
   }
 
+  // Encrypt with shared secret
+  private handleEncrypt(data: any): void{
+    if (!this.approve(data, AccessLevel.ApproveAll)){
+      return;
+    }
+
+    const { id, payload: { encryptedSeedHex, recipientPubkey, message} } = data;
+    const seedHex = this.cryptoService.decryptSeedHex(encryptedSeedHex, this.globalVars.hostname);
+    const encryptedMessage = this.signingService.encryptMessage(seedHex, recipientPubkey, message);
+    this.respond(id, {
+      encryptedMessage
+    });
+  }
+
   private handleDecrypt(data: any): void {
     if (!this.approve(data, AccessLevel.ApproveAll)) {
       return;
     }
 
-    const { id, payload: { encryptedSeedHex, encryptedHexes } } = data;
-    const seedHex = this.cryptoService.decryptSeedHex(encryptedSeedHex, this.globalVars.hostname);
-    const decryptedHexes = this.signingService.decryptMessages(seedHex, encryptedHexes);
+    const seedHex = this.cryptoService.decryptSeedHex(data.payload.encryptedSeedHex, this.globalVars.hostname);
+    const id = data.id;
+
+    let decryptedHexes;
+    if (data.payload.encryptedHexes){
```

Comment:
Can we have two separate methods on signing service and call them here? E.g. `decryptMessages` and `decryptMessagesLegacy`. Will prevent us from needing `string` vs `object` checks in `decryptMessages`. Code duplication is fine because `decryptMessagesLegacy` will be deleted in a few weeks